python/cudf/cudf/_lib/cpp/concatenate.pxd

```diff
@@ -5,16 +5,29 @@ from libcpp.vector cimport vector
 
 from cudf._lib.cpp.column.column cimport column, column_view
 from cudf._lib.cpp.table.table cimport table, table_view
-from rmm._lib.device_buffer cimport device_buffer
+from cudf._lib.cpp.utilities.host_span cimport host_span
 
+from rmm._lib.device_buffer cimport device_buffer
 
 cdef extern from "cudf/concatenate.hpp" namespace "cudf" nogil:
     cdef device_buffer concatenate_masks "cudf::concatenate_masks"(
-        const vector[column_view] columns
+        host_span[column_view] views
+    ) except +
+    cdef unique_ptr[column] concatenate_columns "cudf::concatenate"(
+        host_span[column_view] columns
+    ) except +
+    cdef unique_ptr[table] concatenate_tables "cudf::concatenate"(
+        host_span[table_view] tables
+    ) except +
+
+    # these don't exist in the C++, but construction from vector is possible since
+    # a host_span can be implicitly constructed from a vector, so we add these for convenience
+    cdef device_buffer concatenate_masks "cudf::concatenate_masks"(
+        vector[column_view] views
     ) except +
     cdef unique_ptr[column] concatenate_columns "cudf::concatenate"(
-        const vector[column_view] columns
+        vector[column_view] columns
     ) except +
     cdef unique_ptr[table] concatenate_tables "cudf::concatenate"(
-        const vector[table_view] tables
+        vector[table_view] tables
     ) except +
```

Comment:
@shwina these are the same as the original Cython code for concatenate, and the only ones currently used. So one idea is that I could revert my Cython changes in this PR. We could then wait and add `host_span` in Cython only if we need it...  What do you think?

Reply:
The Cython definitions are nice to have. I suspect at some point we're going to need them, so let's not delete. Agree about keeping the `concat` definitions the same though.